src/consumer/lanes/switch-lanes.ts

```diff
@@ -53,6 +61,20 @@ export default async function switchLanes(consumer: Consumer, switchProps: Switc
   const componentsResults = await mapSeries(succeededComponents, ({ id, componentFromFS, mergeResults }) => {
     return applyVersion(consumer, id, componentFromFS, mergeResults, checkoutProps);
   });
+
+  componentsResults.forEach((cr) => {
+    const existingFilePathsFromModel = cr.applyVersionResult.filesStatus;
+    const filePathsFromFS = succeededComponents.flatMap((sc) => sc.componentFromFS?.files || []);
+
+    filePathsFromFS.forEach((f) => {
+      const key = pathNormalizeToLinux(f.relative);
```

Comment:
looks good. 
about the variable names, I'd suggest being more descriptive. e.g.:
`f` => `file`.
`key` => `filename`.
`cr` => `componentResult` or `result`
`sc` => `succeededComponent`.